src/GLib/signals.jl

```diff
@@ -370,16 +370,34 @@ function g_timeout_add(interval::Integer, cb::Function, user_data::CT) where CT
         0, UInt32(interval), callback, ref, deref)
 end
 
+_g_timeout_add_callback(cb::Function) = Cint(cb())
+function g_timeout_add(interval::Integer, cb::Function)
+    callback = @cfunction(_g_timeout_add_callback, Cint, (Ref{Function},))
+    ref, deref = gc_ref_closure(cb)
+    return ccall((:g_timeout_add_full, libglib),Cint,
+        (Cint, UInt32, Ptr{Cvoid}, Ptr{Cvoid}, Ptr{Cvoid}),
+        0, UInt32(interval), callback, ref, deref)
+end
+
 function g_idle_add(cb::Function, user_data::CT) where CT
```

Comment:
You could now just implement this as
```jl
g_idle_add(cb::Function, user_data) = g_idle_add(() -> cb(Ref(user_data)))
```
and similarly for `g_timeout_add(interval::Integer, cb::Function, user_data)`.